dev/TEAL_opcodes.md

```diff
@@ -1335,6 +1395,68 @@ The notation A,B indicates that A and B are interpreted as a uint128 value, with
 - Availability: v6
 - Mode: Application
 
+## box_create
+
+- Opcode: 0xb9
+- Stack: ..., A: []byte, B: uint64 &rarr; ..., uint64
+- create a box named A, of length B. Fail if A is empty or B exceeds 32,768. Returns 0 if A already existed, else 1
+- Availability: v8
+- Mode: Application
+
+Newly created boxes are filled with 0 bytes. `box_create` will fail if the referenced box already exists with a different size. Otherwise, existing boxes are unchanged by `box_create`.
```

Comment:
What happens if B=0?

Reply:
That's fine, you get a 0 byte box.  You might do this, for example, to implement a set. You use the hash of some item as the box name, but don't actually need to store anything, so you might as well save the 400microalgos extra a 1 byte box would need.